gnd/src/main/java/com/google/android/gnd/ui/offlinebasemap/selector/OfflineBaseMapSelectorFragment.java

```diff
@@ -99,18 +91,32 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
     }
   }
 
+  /** Prepare the map once it's ready. */
   private void onMapReady(MapAdapter map) {
     this.map = map;
   }
 
+  /** Handle a download button click and queue a basemap download. */
   public void onDownloadClick() {
     if (map == null) {
       return;
     }
 
-    viewModel.onDownloadClick(
-        map.getViewport(),
-        map.getCurrentZoomLevel(),
-        getContext().getString(R.string.unnamed_area));
+    viewModel.downloadBaseMap(
+        map.getViewport(), getContext().getString(R.string.unnamed_area));
```

Comment:
Can the VM inject `Resources` and get the string resource itself instead of passing here?

Reply:
Done, thanks!